packages/runtime/runtime-definitions/src/protocol.ts

```diff
@@ -65,3 +69,27 @@ export interface IAttachMessage {
 export type InboundAttachMessage = Omit<IAttachMessage, "snapshot"> & {
 	snapshot: IAttachMessage["snapshot"] | null;
 };
+
+/**
+ * This is the message type that is used within the runtime when processing a sequenced message.
+ * It is the same as ISequencedDocumentMessage, but without the contents and clientSequenceNumbers
+ * which are sent separately. The contents are modified at multiple layers in the stack so having it
+ * separate doesn't require packing and unpacking the entire message.
+ * @alpha
+ * @legacy
+ */
+export type ISequencedRuntimeMessageCore = Omit<
+	ISequencedDocumentMessage,
+	"contents" | "clientSequenceNumber"
+>;
+
+/**
+ * These are the contents of a runtime message as it is processed throughout the stack.
+ * @alpha
+ * @legacy
+ */
+export interface IRuntimeMessageContents {
+	contents: unknown;
+	localOpMetadata: unknown;
+	clientSequenceNumber: number;
```

Comment:
Since you are making changes, can we name it differently? Maybe groupSequence, or something else of this nature.
It feels like there is no need to continue to use here wrong naming and may we can fix it across all layers with your changes.

Reply:
I'd not make changes to IDocumentSequncedMessage (including ungrouping logic - let's leave it for later), but once we convert ops into this format, I'd stop using clientSequenceNumber